sched/task/task_getpid.c

```diff
@@ -95,11 +95,17 @@ pid_t getpid(void)
 
       if (rtcb->task_state == TSTATE_TASK_RUNNING)
         {
+#if !defined(CONFIG_DISABLE_PTHREAD) && defined(CONFIG_SCHED_HAVE_PARENT)
+          /* Yes.. Return the task ID from the main TCB */
```

Comment:
I think that this change should apply even if CONFIG_SCHED_HAVE_PARENT is not defined, should it not?

Reply:
Because the definition use the same condition(https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/sched.h#L475):
```
struct task_group_s
{
#if !defined(CONFIG_DISABLE_PTHREAD) && defined(CONFIG_SCHED_HAVE_PARENT)
  pid_t tg_task;                    /* The ID of the task within the group      */
#endif
```

Reply:
> 
> 
> Because the definition use the same condition(https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/sched.h#L475):
> 
> ```
> struct task_group_s
> {
> #if !defined(CONFIG_DISABLE_PTHREAD) && defined(CONFIG_SCHED_HAVE_PARENT)
>   pid_t tg_task;                    /* The ID of the task within the group      */
> #endif
> ```

That is my point.  The definition should change. tg_task should be initialized and available in all configurations.  Otherwise, getpid() will return a bad value in some configurations.

Reply:
Ok, I will change the definition too.